### PR TITLE
Implemented functionality for find order page ( #1wn444v )

### PR DESCRIPTION
**Pull request by meet-aniket**
Implemented functionality in find order page, I have used findOrder and searchProduct endpoint there and integrated all necessary functionalities but did not get some properties there.

### CODE REVIEW

File: changelogs/unreleased/-1wn444v.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: Implemented Open order page functionality and created store for that that (#1wn444v)
```
Comment (ymaheshwari1) [inline changelogs/unreleased/-1wn444v.yml]: Remove ticket id from the title

File: src/components/OrderCard.vue [hunk 2]
```diff
@@ -0,0 +1,115 @@
+<template>
+    <div class="order-header">
```
Comment (ymaheshwari1) [inline src/components/OrderCard.vue]: Improve indentation

File: src/components/OrderCard.vue [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  components: {
+    IonBadge,
+    IonChip,
+    IonIcon,
+    IonItem,
+    IonLabel,
+    IonNote,
+    OpenItemCard,
+  },
+	props: ['order'],
```
Comment (ymaheshwari1) [inline src/components/OrderCard.vue]: Improve indentation

File: src/router/index.ts [hunk 4]
```diff
@@ -26,6 +28,18 @@ const routes: Array<RouteRecordRaw> = [
...
+    name: 'Order',
+    component: Order,
+    beforeEnter: authGuard
+  },
+  // {
+  //   path: '/product',
+  //   name: 'ProductInventory',
+  //   component: ProductInventory,
+  //   beforeEnter: authGuard
+  // },
```
Comment (ymaheshwari1) [inline src/router/index.ts]: Remove this if not needed.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Orders.vue [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+  IonList,
+  IonListHeader,
+  IonNote,
+  IonPage,
+  IonSearchbar,
+  IonSelect,
+  IonThumbnail,
+  IonTitle,
+  IonToolbar,
+  IonSelectOption,
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:193]: Arrange the component import alphabetically

File: src/main.ts [hunk 6]
```diff
@@ -57,7 +57,7 @@ app.config.globalProperties.$filters = {
       externalId = externalIdentificationSplit[1] ? externalIdentificationSplit[1] : '';
     }
     return externalId;
-  },
+  },
```
Comment (ymaheshwari1) [inline src/main.ts]: Remove this unwanted change

File: src/store/modules/product/actions.ts [hunk 7]
```diff
@@ -40,12 +40,48 @@ const actions: ActionTree<ProductState, RootState> = {
       // Remove added loader only when new query and not the infinite scroll
       if (payload.viewIndex === 0) emitter.emit("dismissLoader");
     } catch(error){
-      console.log(error)
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:43]: Why we have removed this console statement?

File: src/store/modules/product/ProductState.ts [hunk 8]
```diff
@@ -1,6 +1,8 @@
 export default interface ProductState {
+  cached: any;  
   products: {
     list: any;
     total: number;
   }
+  productsInformation: any;
```
Comment (ymaheshwari1) [inline src/store/modules/product/ProductState.ts]: I think this state is not needed.

File: src/store/modules/order/actions.ts [hunk 9]
```diff
@@ -27,6 +30,22 @@ const actions: ActionTree<OrderState, RootState> = {
     }
     return resp;
   },
+
+  // Add Product information in cache
+
+  async getProductInformation(context, { orders }){
```
Comment (ymaheshwari1) [inline src/store/modules/order/actions.ts]: Remove duplicate declaration of the same function. You have defined the `getProductInformation` here in the order action and also in the product action.

File: src/store/modules/product/index.ts [hunk 10]
```diff
@@ -8,10 +8,12 @@ import RootState from '../../RootState'
 const productModule: Module<ProductState, RootState> = {
     namespaced: true,
     state: {
+      cached: {},
       products: {
         list: {},
         total: 0
-      }
+      },
+      productsInformation: {}
```
Comment (ymaheshwari1) [inline src/store/modules/product/index.ts]: Remove the `productsInformation` state.

File: src/store/modules/product/mutations.ts [hunk 11]
```diff
@@ -6,6 +6,14 @@ const mutations: MutationTree <ProductState> = {
   [types.PRODUCT_SEARCH_UPDATED] (state, payload) {
     state.products.list = payload.products;
     state.products.total = payload.totalProductsCount;
-  }
+  },
+  [types.PRODUCT_ADD_TO_CACHED_MULTIPLE] (state, payload) {
+      // TODO
+      if (payload.products) {
```
Comment (ymaheshwari1) [inline src/store/modules/product/mutations.ts]: Improve indentation

File: src/store/modules/product/mutations.ts [hunk 12]
```diff
@@ -6,6 +6,14 @@ const mutations: MutationTree <ProductState> = {
   [types.PRODUCT_SEARCH_UPDATED] (state, payload) {
     state.products.list = payload.products;
     state.products.total = payload.totalProductsCount;
-  }
+  },
+  [types.PRODUCT_ADD_TO_CACHED_MULTIPLE] (state, payload) {
+      // TODO
```
Comment (ymaheshwari1) [inline src/store/modules/product/mutations.ts]: Why here a TODO statement?

File: src/views/Orders.vue [hunk 13]
```diff
@@ -0,0 +1,347 @@
...
+  computed: {
+    ...mapGetters({
+      orders: 'order/getList',
+      getProduct: 'product/getProduct',
+      currentFacilityId: 'user/getCurrentFacility'
+    })
+  },
+
+  methods: {
+    async getOrders(vSize, vIndex){
```
Comment (ymaheshwari1) [inline src/views/Orders.vue]: Make `vSize` and `vIndex` optional.

File: src/views/Orders.vue [hunk 14]
```diff
@@ -0,0 +1,347 @@
...
+      await Clipboard.write({
+        string: text
+      }).then(() => {
+        showToast(this.$t('Copied', { text }));
+      })
+    },
+
+  },
+  mounted() {
+    this.getOrders(10, 0);
```
Comment (ymaheshwari1) [inline src/views/Orders.vue]: We won't pass the view size and view index from the mounted hook

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 6 above)
Comment (meet-aniket) [reply src/main.ts]: Removed Sir.

File: src/store/modules/product/actions.ts (hunk 7 above)
Comment (meet-aniket) [reply src/store/modules/product/actions.ts:43]: Aditya sharma Sir instructed me to remove all console.log from code before making PR.

File: src/store/modules/product/ProductState.ts (hunk 8 above)
Comment (meet-aniket) [reply src/store/modules/product/ProductState.ts]: Sir according to me this statement is removed in recent commits please check those

File: src/store/modules/order/actions.ts (hunk 9 above)
Comment (meet-aniket) [reply src/store/modules/order/actions.ts]: Okay sir will update

File: src/views/Orders.vue (hunk 14 above)
Comment (meet-aniket) [reply src/views/Orders.vue]: Okay, Sir

Comment (meet-aniket) [reply src/views/Orders.vue:261]: Removed but this was optional for testing purpose

File: src/views/Orders.vue (hunk 13 above)
Comment (meet-aniket) [reply src/views/Orders.vue]: Okey sir

File: src/store/modules/product/mutations.ts (hunk 12 above)
Comment (meet-aniket) [reply src/store/modules/product/mutations.ts]: Removed Sir

File: src/store/modules/product/mutations.ts (hunk 11 above)
Comment (meet-aniket) [reply src/store/modules/product/mutations.ts]: Improved sir

File: src/store/modules/product/index.ts (hunk 10 above)
Comment (meet-aniket) [reply src/store/modules/product/index.ts]: Already removed Sir

File: changelogs/unreleased/-1wn444v.yml (hunk 1 above)
Comment (meet-aniket) [reply changelogs/unreleased/-1wn444v.yml]: Removed Sir

File: src/components/OrderCard.vue (hunk 2 above)
Comment (meet-aniket) [reply src/components/OrderCard.vue]: Improved Sir

File: src/components/OrderCard.vue (hunk 3 above)
Comment (meet-aniket) [reply src/components/OrderCard.vue]: Improved sir

File: src/router/index.ts (hunk 4 above)
Comment (meet-aniket) [reply src/router/index.ts]: Removed Sir

File: src/views/Orders.vue (hunk 5 above)
Comment (meet-aniket) [reply src/views/Orders.vue:193]: Arranged Sir

File: src/store/modules/product/actions.ts (hunk 7 above)
Comment (ymaheshwari1) [reply src/store/modules/product/actions.ts:43]: But this console statement is not added by you and it's used to display the error message

File: src/views/Orders.vue (hunk 13 above)
Comment (meet-aniket) [reply src/views/Orders.vue]: Sir made them optional

**Comment by meet-aniket** [comment]
Screenshot from 2021-12-21 18-18-31
Screenshot from 2021-12-21 18-17-32

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]
Remove package-lock.json file changes.

**Comment by meet-aniket** [comment]
This pull request can not be accessed by me for making further changes so this is why I am closing this pull request.

**Pull request closed without merging**